extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -750,6 +781,45 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching a new creative and rendering it. This leaves
+   * the current creative displayed until the next one is ready.
+   *
+   * @param {function()} refreshEndCallback When called, this function will
+   *   restart the refresh cycle.
+   * @return {Promise} A promise that resolves when all asynchronous portions of
+   *   the refresh function complete. This is particularly handy for testing.
+   */
+  refresh(refreshEndCallback) {
+    return new Promise(resolve => {
+      dev().assert(!this.isRefreshing);
+      this.isRefreshing = true;
+      this.tearDownSlot();
+      this.initiateAdRequest();
+      const promiseId = this.promiseId_;
+      this.adPromise_.then(() => {
+        if (!this.isRefreshing || promiseId != this.promiseId_) {
+          // If this refresh cycle was canceled, such as in a no-content
+          // response case, keep showing the old creative.
+          refreshEndCallback();
+          return;
+        }
+        this.getVsync().mutate(() => {
```

Comment:
We discussed both options, changing priority and using timer. It seems to me that messing with the priority to get a 1 second delay seems like roundabout way of doing it when we can just use a timer. 

> after ad request, if valid creative (which I'm not sure you're asserting)

If the creative would force a collapse, then isRefreshing would be set to false, short circuiting the function early. Also, as previously mentioned, timerFor returns a string cancellation id, not a promise.